modules/react/popup/lib/PopupCloseIcon.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+import {xIcon} from '@workday/canvas-system-icons-web';
+import {space} from '@workday/canvas-kit-react/tokens';
+import {
+  styled,
+  createComponent,
+  StyledType,
+  useModelContext,
+  createHook,
+  changeFocus,
+} from '@workday/canvas-kit-react/common';
+import {IconButton, IconButtonProps} from '@workday/canvas-kit-react/button';
+
+import {PopupModel} from './usePopupModel';
+import {PopupModelContext} from './Popup';
+
+const closeIconSpacing = space.xs;
+const closeIconSpacingSmall = 10;
+
+export interface PopupCloseIconProps extends IconButtonProps {
+  /**
+   * Optionally pass a model directly to this component. Default is to implicitly use the same
+   * model as the container component which uses React context. Only use this for advanced use-cases
+   */
+  model?: PopupModel;
+}
+
+const StyledCloseIcon = styled(IconButton)<StyledType & IconButtonProps>(
+  {
+    position: 'absolute',
+  },
+  ({size}) => ({
+    right: size === 'small' ? closeIconSpacingSmall : closeIconSpacing,
+    top: size === 'small' ? closeIconSpacingSmall : closeIconSpacing,
+  })
+);
+
+const useCloseButton = createHook(({events, state}: PopupModel) => {
+  return {
+    onClick: () => {
+      events.hide();
+
+      // delay until after state change have happened and changes are flushed to the DOM
+      requestAnimationFrame(() => {
+        if (state.willReturnFocus?.current) {
+          changeFocus((state.returnFocusRef || state.targetRef).current);
+        }
+      });
+    },
+  };
+});
+
+export const PopupCloseIcon = createComponent('button')({
```

Comment:
We use element strings as the default when the component handles its own styling. We use components as the default when the component doesn't handle styling (behavior-only components).

I guess this is kind of middle ground. It does seem like it should be `IconButton`, but the component does add absolution positioning to get the icon in the right spot on the Popup. Before this was handled by a `div` element that did the positioning. I didn't want to do the `div` route because it meant nobody could change the location if they really needed to.

I guess the question is what would people expect to happen why they pass an `as`? Would they want to change the rendered element or the rendered component? `IconButton` still works perfectly fine since we're really rendering a `StyledIconButton` and the `as` is passed down to `styled`. `StyledIconButton` is still a `styled(IconButton)`, so changing the rendered component doesn't change how it looks. For example, `<Popup.CloseIcon as={DeleteButton}>` still renders as an `IconButton`. Emotion will merge the styles together and the `IconButton` styles will win (this example makes for a long css prop list).